File: query_primer/query_primer__sort_find_multi.cs
using System.Linq;
using static System.Console;

internal class Program
{
    static void Main()
    {
        int o = 0;
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();
        var a = new System.Collections.Generic.List<int>( n[0] + n[1] ) { n[2] };

        SetOut( new System.IO.StreamWriter( OpenStandardOutput() ) { AutoFlush = false } );
        foreach( int _ in Enumerable.Range( 0, n[0] ) ) {
            a.Add( int.Parse( ReadLine() ) );
        }
        o = a.OrderBy( x => x ).ToList().IndexOf( n[2] ) + 1;
        foreach( int _ in Enumerable.Range( 0, n[1] ) ) {
            string[] s = ReadLine().Split();
            var c = System.StringComparison.Ordinal;

            if( string.Compare( s[0], "join", c ) == 0 && int.Parse( s[1] ) < n[2] ) {
                o++;
            } else if( string.Compare( s[0], "sorting", c ) == 0 ) {
                WriteLine( o );
            }
        }
        Out.Flush();
        return;
    }
}

File: query_primer/query_primer__sort_find_multi.py
n, k, p = list(map(int, input().split()))
l = [int(input()) for _ in range(n)]

l.append(p)

o = sorted(l).index(p) + 1

for _ in range(k):
    s = input().split()

    if s[0] == "join" and int(s[1]) < p:
        o += 1
    elif s[0] == "sorting":
        print(o)
